feat(register)!: network only store ops list

maqi opened the pull request:
BREAKING CHANGE!

crdt MerkleReg only stored in client locally
node side only store ops list (signed by owner)



<!--
### Footer (Hidden from GitHub view)
This template uses [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/). Please ensure your commit messages follow these guidelines for clarity and consistency.

Commit messages should be verified using [commitlint](https://commitlint.js.org/#/).

Commit Message Format:
<type>[optional scope]: <description>
[optional body]
[optional footer(s)]

Common Types:
- `feat`
- `fix`
- `docs`
- `style`
- `refactor`
- `perf`
- `test`
- `build`
- `ci`
- `chore`
- `revert`
-->

bochaco commented on sn_interface/src/messaging/data/mod.rs:
```diff
@@ -239,10 +223,10 @@ impl CmdResponse {
-    pub fn is_success(&self) -> bool {
-        self.result().is_ok()
-    }
+    // /// Returns true if the result returned is a success or not
```
shall we remove this fn too?

bochaco commented on sn_interface/src/messaging/system/dkg.rs:
```diff
@@ -97,7 +76,7 @@ impl DkgSessionId {
-    pub fn contains_elder(&self, elder: XorName) -> bool {
-        self.elder_names().any(|e| e == elder)
-    }
+    // pub fn contains_elder(&self, elder: XorName) -> bool {
```
shall we remove it too?

bochaco commented on sn_interface/src/network_knowledge/mod.rs:
```diff
@@ -530,34 +524,34 @@ impl NetworkKnowledge {
-        self.section_peers.num_of_members()
-    }
+    // /// Returns current section size, i.e. number of peers in the section.
+    // pub fn section_size(&self) -> usize {
```
another to remove?

maqi committed:
feat(register)!: network only store ops list

BREAKING CHANGE!

crdt MerkleReg only stored in client locally
node side only store ops list (signed by owner)

b-zee commented on autonomi/src/client/registers.rs:
```diff
@@ -42,27 +43,68 @@ pub enum RegisterError {
+        Ok(register)
+    }
+
+    pub fn write_atop(&mut self, entry: &[u8], owner: &SecretKey) -> Result<(), RegisterError> {
```
Lets not make these `pub`. I want to keep the public interface simple.

b-zee commented on autonomi/src/client/registers.rs:
```diff
@@ -42,27 +43,68 @@ pub enum RegisterError {
             .collect()
     }
+
+    pub fn new(
```
```suggestion
    fn new(
```

b-zee commented on sn_registers/src/reg_crdt.rs:
```diff
@@ -9,17 +9,21 @@
 /// Register data type as a CRDT with Access Control
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Hash, PartialOrd)]
-pub(crate) struct RegisterCrdt {
+pub struct RegisterCrdt {
```
Is there a specific motivation for making these items `pub`?

happybeing replied:
@b-zee I think that's needed so that the API can expose the MerkleReg. That's necessary for access to the history, and without it usefulness of Registers becomes very limited. Not useful really!

You can look at/use the `awe inspect-register` command and implementation for some of the things that are needed.

b-zee replied:
Yes, we do want to expose the underlying register. But that would be the `MerkleReg` type. Our `RegisterCrdt` is not something we want to promise anything about for the future.
